src/lib.rs

```diff
@@ -146,19 +146,22 @@ pub fn eval_beat(cmds: &[Cmd], t: f64) -> Result<f64, ()> {
                 }
             }
             Arr(size) => {
-                // We need to pop index + size values, so our stack needs to be
-                // atleast size + 1 elements long
-                if size + 1 > stack.len() || size == 0 { 
-                    stack.push(0.0)
+                let index = stack.pop().ok_or(())? as i32;
+                // We need to pop size values, so our stack needs to be
+                // atleast size elements long. Note that split_off panics if we
+                // exceed the length of the vector, so we need this if guard.
+                if size > stack.len() { 
+                    return Err(())
+                } else if size == 0 {
+                    stack.push(0.0);
                 } else {
                     // We want to split off from the end, so we must subtract
                     // here. Note that we add one for the index.
-                    let split_index = stack.len() - (size + 1);
+                    let split_index = stack.len() - size;
                     let mut vec = stack.split_off(split_index);
                     let size = size as i32; // too lazy to keep tying as i32
```

Comment:
I don't think this `as` is worth commenting on, although I think it should go to `i64` not `i32`.